dotnet/src/SemanticKernel.Abstractions/Memory/RecordAttributes/VectorStoreRecordDataAttribute.cs

```diff
@@ -22,4 +22,10 @@ public sealed class VectorStoreRecordDataAttribute : Attribute
     /// Gets or sets the name of the property that contains the embedding for this data field.
     /// </summary>
     public string? EmbeddingPropertyName { get; init; }
+
+    /// <summary>
+    /// Gets or sets an optional name to use for the property in storage, if different from the property name.
+    /// E.g. the property name might be "MyProperty" but the storage name might be "my_property".
+    /// </summary>
+    public string? StoragePropertyName { get; set; }
```

Comment:
`EmbeddingPropertyName` and `StoragePropertyName` look very similar, but they have different purposes - `EmbeddingPropertyName` is used to reference another property in same C# type to create an embedding-data relationship, while `StoragePropertyName` is used for field naming in database. Taking into account that both properties live in the same attribute, I'm wondering if it won't be confusing for users which one to use without reading XML documentation.

As far as I understand, we can't use `JsonPropertyName` attribute in all connectors, and we need to provide custom naming mechanism. Should we create similar attribute for name in this case? At least, from the usage perspective, it will be similar.

With current implementation:
```csharp
// JSON way to provide name
[JsonPropertyName("parking_is_included")]
[VectorStoreRecordData]
public bool ParkingIncluded { get; init; }

// Our way to provide name
[VectorStoreRecordData(StoragePropertyName = "parking_is_included")]
public bool ParkingIncluded { get; init; }
```

With custom property name attribute:
```csharp
// JSON way to provide name
[JsonPropertyName("parking_is_included")]
[VectorStoreRecordData]
public bool ParkingIncluded { get; init; }

// Our way to provide name
[RecordPropertyName("parking_is_included")]
[VectorStoreRecordData]
public bool ParkingIncluded { get; init; }
```

Also, with custom property name attribute you won't have to put `StoragePropertyName` in every record attribute available at the moment (you can use inheritance to resolve this problem as well). In general, it feels like record attributes and property name should be detached, in the same way how it works with `JsonPropertyName` at the moment.

Reply:
Thanks for the comment.  I like what you are suggesting, but we also have the separate non-attribute configuration, and I was trying to keep those in sync.  That has a StoragePropertyName property on each of the field config classes too. It would be painful from a consumer perspective to separate that out.  So we have two options.  We diverge, where we have multiple attributes and one config object per property, which might be ok.  Or we try and keep it all the same and don't separate out the attributes.

Reply:
As discussed, will keep the config and attributes the same for now.